src/field_theory/finite/galois_field.lean

```diff
@@ -151,17 +150,14 @@ by rw [←splits_id_iff_splits, polynomial.map_sub, polynomial.map_pow, map_X, s
 
 lemma is_splitting_field_of_card_eq (h : fintype.card K = p ^ n) :
   is_splitting_field (zmod p) K (X ^ (p ^ n) - X) :=
-{ splits := by { rw ← h, exact splits_X_pow_card_sub_X p },
-  adjoin_roots :=
-  begin
-    have hne : n ≠ 0,
-    { rintro rfl, rw [pow_zero, fintype.card_eq_one_iff_nonempty_unique] at h,
-      cases h, resetI, exact false_of_nontrivial_of_subsingleton K },
-    refine algebra.eq_top_iff.mpr (λ x, algebra.subset_adjoin _),
-    rw [polynomial.map_sub, polynomial.map_pow, map_X, finset.mem_coe, multiset.mem_to_finset,
-        mem_roots, is_root.def, eval_sub, eval_pow, eval_X, ← h, finite_field.pow_card, sub_self],
-    exact finite_field.X_pow_card_pow_sub_X_ne_zero K hne (fact.out _)
-  end }
+h ▸ finite_field.has_sub.sub.polynomial.is_splitting_field K p
+
+instance : is_galois (zmod p) K :=
+is_galois.of_separable_splitting_field (galois_poly_separable p (fintype.card K) begin
+  haveI := char_p_of_injective_algebra_map (algebra_map (zmod p) K).injective p,
```

Comment:
```suggestion
is_galois.of_separable_splitting_field (galois_poly_separable p (fintype.card K)
begin
  haveI := char_p_of_injective_algebra_map (algebra_map (zmod p) K).injective p,
```

maybe i'm being picky but I don't like that `begin` just hanging there.

also, can we make that `char_p_of_injective...` an instance for `division_ring`s?

Reply:
I tried adding
```
instance char_p_of_injective_algebra_map' {R A : Type*} [field R] [semiring A] [algebra R A]
  [nontrivial A] (p : ℕ) [char_p R p] : char_p A p :=
char_p_of_injective_algebra_map (algebra_map R A).injective p
```
but it caused problems.

Reply:
That can't be an instance because lean can't work out R

Reply:
But you could add the instance for `R = zmod p`

Reply:
Still causing problems. I think the issue is that it causes a loop with `zmod.algebra`.

Reply:
I don't see an easy way to keep this as an instance, as you say the `[algebra R A]` parameter combined with `zmod.algebra` will basically always cause loops. So let's add your original `char_p_of_injective_algebra_map'` as a `def`.